llama_hub/airbyte_hubspot/README.md

```diff
@@ -0,0 +1,58 @@
+# Airbyte Hubspot Loader
+
+The Airbyte Hubspot Loader allows you to access different Hubspot objects.
+
+## Installation
+
+* Install llama_hub: `pip install llama_hub`
+* Install the hubspot source: `pip install airbyte-source-hubspot`
+
+## Usage
+
+Here's an example usage of the AirbyteHubspotReader.
+
+```python
+from llama_hub.airbyte_hubspot.base import AirbyteHubspotReader
+
+hubspot_config = {
+    # ...
+}
+reader = AirbyteHubspotReader(config=hubspot_config)
+documents = reader.load_data(stream_name="products")
+```
+
+## Configuration
+
+Check out the [Airbyte documentation page](https://docs.airbyte.com/integrations/sources/hubspot/) for details about how to configure the reader.
+The JSON schema the config object should adhere to can be found on Github: [https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-hubspot/source_hubspot/spec.yaml](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-hubspot/source_hubspot/spec.yaml).
+
+The general shape looks like this:
+```python
+{
+  "start_date": "<date from which to start retrieving records from in ISO format, e.g. 2020-10-20T00:00:00Z>",
+  "credentials": {
+    "credentials_title": "Private App Credentials",
+    "access_token": "<access token of your private app>"
+  }
+}
+```
+
+By default all fields are stored as metadata in the documents and the text is set to an empty string. Construct the text of the document by transforming the documents returned by the reader:
+```python
+for doc in documents:
+  doc.text = doc.extra_info["title"]
```

Comment:
So to expand on the other comment further:

If I'm thinking about a hubspot use case, it may be let me do some Q&A over the previous notes and emails I've had with a partner.

The content of the notes and emails are probably what we want to be chunked and retrieved over. If we're only embedding each note individually without any additional text splitting, then we're OK because we embed all of the metadata by default.

However, if we want to start text splitting, then the problem arises, because 1. the splitting will only happen on doc.text, but 2. somewhat even more problematic, every split will have a complete copy of the document's metadata.

3000 words is a lot of text (definitely bigger than most interaction notes and emails) so we may be able to get away with it for now, but I think longer run we'll want a different document class to handle this kind of use case.

There could be a lot of interesting things here, because after the loader loads it in in this more structured way, the user or the application designer could say: I want my coworkers' Notion comments to be chunked alongside the content of the doc, or no, I don't want the comments to split alongside the doc and it's just a matter of adjusting a list for the fields to include in the splitting.

Reply:
the other alternative is to dump *everything* into text actually (so naive text dump), and if the user wants to have more fine-grained metadata specifications they use this argument

Reply:
I like the record handler plus dumping everything in there by default, seems like the easiest way to get started. The text could get really messy for some streams, but that's probably ok